DataReader.py

```diff
@@ -0,0 +1,99 @@
+import datetime
+import csv
+import glob
+import collections
+
+from Weather import Weather
+from View import View
+
+
+class data_reader:
+
+	annual_report = {}
+	annual_max_temp = {}
+	year = ""
+
+
+	def get_max(self, max_value, to_compare):
+		
+		if max_value < to_compare:
+			return True
+		
+		return False
+
+	def get_min(self, min_value, to_compare):
+		
+		if min_value < to_compare and min_value > 0:
+			return min_value
+		
+		return to_compare
+
+	def retreive_dir(self, dir_path):
+		if not glob.glob(dir_path):
+			print 'No files were found!'
+			return
+
+		for textfile in glob.glob(dir_path + "/*txt" ):
+			weather = self.read_file(textfile)
+			if weather:
+				self.compare_data(weather)
+
+	def compare_data(self, weather):		
```

Comment:
Avoid these second comparisons. Firstly it compares values for the same month to get appropriate values for the month

> Happening in `read_file`

and then it compares with appropriate values of previously processed months

> Happening in `compare_data`

It can be merged to one step.